analyzer/src/test/kotlin/PackageJsonUtilTest.kt

```diff
@@ -0,0 +1,178 @@
+                if (!matchers.isEmpty())
+                """
+                {
+                   "workspaces" : ${matchers.joinToString (prefix = "[\"", separator = "\",\"", postfix = "\"]")}
+                }
+                """.trimIndent()
+                else "{}"
+
+        private fun mapDefinitionFiles(definitionFiles: Collection<File>) =
+                mapDefinitionFilesForNpm(definitionFiles).plus(mapDefinitionFilesForYarn(definitionFiles))
+    }
+
+    init {
+
+        "given no npm lockfile present, hasNpmLockFile returns false" {
+            setupProject(path = "a")
+
+            hasNpmLockFile("a") shouldBe false
+        }
+
+        "given npm lockfile present, hasNpmLockFile returns true" {
+            setupProject(path = "a", hasNpmLockFile = true)
+
+            hasNpmLockFile("a") shouldBe true
+        }
+
+        "given no yarn lockfile present, hasYarnLockFile returns false" {
+            setupProject(path = "a")
+
+            hasNpmLockFile("a") shouldBe false
+        }
+
+        "given yarn lockfile present, hasYarnLockFile returns true" {
+            setupProject(path = "a", hasYarnLockFile = true)
+
+            hasYarnLockFile("a") shouldBe true
+        }
+
+        "given project with yarn and npm lockfile, definition file is mapped for yarn only" {
+            setupProject(path = "a", hasNpmLockFile = true, hasYarnLockFile = true)
+
+            mapDefinitionFilesForNpm(definitionFiles).shouldBeEmpty()
+            mapDefinitionFilesForYarn(definitionFiles) shouldContainExactly
+                    absolutePaths(
+                    "a/package.json")
+        }
+
+        "given project with no lockfile, definition file is mapped for npm only" {
+            setupProject(path = "a")
+
+            mapDefinitionFilesForNpm(definitionFiles) shouldContainExactly
+                    absolutePaths("a/package.json")
+            mapDefinitionFilesForYarn(definitionFiles).shouldBeEmpty()
+        }
+
+        "given project matched exactly by workspace, matched project is not mapped" {
+            setupProject(path = "a", matchers = listOf("b"))
+            setupProject(path = "a/b")
+            setupProject(path = "a/c")
+
+            mapDefinitionFiles(definitionFiles) shouldContainExactlyInAnyOrder
+                    absolutePaths(
+                            "a/package.json",
+                            "a/c/package.json")
+        }
+
+        "given projects matched via * by workspace, matched projects are not mapped" {
+            setupProject(path = "a", matchers = listOf("*", "*/f"))
+            setupProject(path = "a/b")
+            setupProject(path = "a/c")
+            setupProject(path = "a/d/e")
+            setupProject(path = "a/d/f")
+
+            mapDefinitionFiles(definitionFiles) shouldContainExactlyInAnyOrder
+                    absolutePaths(
+                            "a/package.json",
+                            "a/d/e/package.json")
```

Comment:
When putting arguments in separate lines the closing paranthesis should go to next line according to the Kotlin coding conventions (same in the next test):
https://kotlinlang.org/docs/reference/coding-conventions.html#method-call-formatting